frontend/server/src/SecurityTools.php

```diff
@@ -248,13 +248,65 @@ public static function getGitserverAuthorizationToken(
      * gitserver that is valid for a single problem for 5 minutes.
      *
      * @param array<string, string> $claims
-     * @param string $subject
+     * @param \OmegaUp\DAO\VO\Identities $subject
      * @return string The Bearer authorization token.
      */
     public static function getCourseCloneAuthorizationToken(
         array $claims,
-        string $subject
+        \OmegaUp\DAO\VO\Identities $subject
     ): string {
+        $secretKey = self::getCourseCloneSecretKey();
+        $token = (new \ParagonIE\Paseto\Builder())
+            ->setKey($secretKey)
+            ->setVersion(new \ParagonIE\Paseto\Protocol\Version2())
+            ->setPurpose(\ParagonIE\Paseto\Purpose::local())
+            ->setExpiration(
+                (\DateTime::createFromFormat(
+                    'U',
+                    \OmegaUp\Time::get()
+                ))->add(
+                    new \DateInterval(
+                        'P7D'
+                    )
+                )
+            )
+            ->setIssuer($subject->username)
+            ->setSubject(strval($subject->user_id))
+            ->setClaims($claims);
+        return $token->toString();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getDecodedCloneCourseToken(
+        string $token,
+        string $courseAlias
+    ): array {
+        $parser = \ParagonIE\Paseto\Parser()::getLocal(
+            self::getCourseCloneSecretKey(),
+            \ParagonIE\Paseto\ProtocolCollection::v2()
+        )
+            ->addRule(
+                new \ParagonIE\Paseto\Rules\ValidAt(
+                    (new \DateTime())->setTimestamp(\OmegaUp\Time::get())
+                )
+            )
+            ->addRule(new ClaimRule('course', $courseAlias))
+            ->addRule(new ClaimRule('permissions', 'clone'));
+
+        $parsedToken = $parser->parse($token, /*$skipValidation=*/true);
+        $claims = $parsedToken->getClaims();
+        if (!$parsedToken->validate()) {
+            throw new \ParagonIE\Paseto\JsonToken($claims);
```

Comment:
```suggestion
            throw new \OmegaUp\Exceptions\TokenDecodeException($claims);
```

hay que crear una clase nueva porque `\ParagonIE\Paseto\JsonToken` no es una excepción.

Reply:
Listo